product/src/main/java/com/yas/product/service/ProductService.java

```diff
@@ -521,15 +521,14 @@ private static void setValuesForVariantExisting(
             variantInDb.setSku(variant.sku());
             variantInDb.setGtin(variant.gtin());
             variantInDb.setPrice(variant.price());
-            List<ProductImage> productImages = variantInDb.getProductImages();
+            productImageRepository.deleteByProductId(variant.id());
             if (CollectionUtils.isNotEmpty(variant.productImageIds())) {
```

Comment:
Hi @khanhduzz 

Would it be possible to apply the early return/exit pattern here? For instance, instead of checking if the list is not empty, you could check if it is empty and return immediately if that’s the case. This would help reduce the current nesting level of the if condition to just one.

Just a suggestion to consider for improving readability!